web/src/pages/apd/new/ApdNew.js

```diff
@@ -0,0 +1,210 @@
+import PropType from 'prop-types';
+import React, { Fragment, useState } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { connect } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { Alert, Button, ChoiceList, TextField } from '@cmsgov/design-system';
+import { createApd } from '../../../redux/actions/app';
+
+import HitechView from './HitechView';
+import MmisView from './MmisView';
+
+import Joi from 'joi';
+
+const createApdSchema = Joi.object({
+  apdType: Joi.valid('hitech', 'mmis').required(),
+  name: Joi.string().min(1).required(),
+  years: Joi.array().min(3).required()
+});
+
+const ApdNew = () => {
+  ApdNew.displayName = 'ApdNew';
+  const history = useHistory();
+  const [apdType, setApdType] = useState(false);
+  const [{ createApd: create }] = useState(false);
+  const [name, setName] = useState(false);
+  const [setIsLoading] = useState(false);
+
+  const createNew = () => {
+    setIsLoading(true);
+    create();
+  };
+
+  const { control, trigger } = useForm({
+    defaultValues: {
+      apdType
+    },
+    mode: 'onBlur',
+    reValidateMode: 'onBlur'
+  });
+
+  const thisFFY = (() => {
+    const year = new Date().getFullYear();
+
+    // Federal fiscal year starts October 1,
+    // but Javascript months start with 0 for
+    // some reason, so October is month 9.
+    if (new Date().getMonth() > 8) {
+      return year + 1;
+    }
+    return year;
+  })();
+
+  const yearOptions = [thisFFY, thisFFY + 1, thisFFY + 2].map(y => `${y}`);
+  const years = yearOptions.slice(0, 2);
+
+  const yearChoices = yearOptions.map((year, index) => ({
+    defaultChecked: years.includes(year),
+    name: `years[${index}].year`,
+    label: year,
+    value: year
+  }));
+
+  const initialState = {
+    apdType,
+    name,
+    years: years
+  };
+
+  const setYears = value => {
+    try {
+      let index = years.indexOf(value);
+      if (index < 0) {
+        let newYears = years.push(value);
+        initialState.years = newYears;
```

Comment:
you don't need initialState, you can let react-hook-form store the values, you should use this
```
setValue('years', newYears, { shouldValidate: true })
```
to set the value